forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/ApplicationPinDeviceAuthenticator.kt

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2022 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+package org.forgerock.android.auth.devicebind
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.fragment.app.FragmentActivity
+import androidx.security.crypto.EncryptedFile
+import androidx.security.crypto.MasterKeys
+import org.forgerock.android.auth.InitProvider
+import org.spongycastle.asn1.x500.X500Name
+import org.spongycastle.asn1.x509.AlgorithmIdentifier
+import org.spongycastle.asn1.x509.SubjectPublicKeyInfo
+import org.spongycastle.cert.X509v1CertificateBuilder
+import org.spongycastle.cert.jcajce.JcaX509CertificateConverter
+import org.spongycastle.crypto.params.AsymmetricKeyParameter
+import org.spongycastle.crypto.util.PrivateKeyFactory
+import org.spongycastle.operator.DefaultDigestAlgorithmIdentifierFinder
+import org.spongycastle.operator.DefaultSignatureAlgorithmIdentifierFinder
+import org.spongycastle.operator.bc.BcRSAContentSignerBuilder
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+import java.math.BigInteger
+import java.security.KeyStore
+import java.security.KeyStore.PasswordProtection
+import java.security.KeyStore.PrivateKeyEntry
+import java.security.PrivateKey
+import java.security.UnrecoverableKeyException
+import java.security.cert.X509Certificate
+import java.security.spec.RSAKeyGenParameterSpec
+import java.util.*
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * Device Authenticator which use Application PIN to secure device cryptography keys
+ */
+open class ApplicationPinDeviceAuthenticator(private val context: Context) : CryptoAware, DeviceAuthenticator {
+
+    internal lateinit var keyAware: KeyAware
+
+    private val PIN_SUFFIX = "_PIN"
```

Comment:
PIN_SUFFIX - upper cases and underscore are for enum.  private property and const are lowercase

Reply:
updated.